pkg/local_object_storage/metabase/version.go

```diff
@@ -278,3 +294,57 @@ func migrateFrom4Version(db *DB) error {
 		return updateVersion(tx, 5)
 	})
 }
+
+// fixMiddleObjectRoots removes root object marks for middle split objects, they
+// were erroneously treated as root ones before the fix.
+func fixMiddleObjectRoots(l *zap.Logger, tx *bbolt.Tx, b *bbolt.Bucket, cnr cid.ID, after []byte, limit uint) (uint, []byte, error) {
+	var (
+		c            = b.Cursor()
+		k            []byte
+		oidKey       = make([]byte, addressKeySize)
+		attrIDPrefix = []byte{metaPrefixAttrIDPlain}
+		queue        = make([]oid.ID, 0, limit)
+	)
+	attrIDPrefix = append(attrIDPrefix, object.FilterRoot...)
+	attrIDPrefix = append(attrIDPrefix, objectcore.MetaAttributeDelimiter...)
+	attrIDPrefix = append(attrIDPrefix, binPropMarker...)
+	attrIDPrefix = append(attrIDPrefix, objectcore.MetaAttributeDelimiter...)
+	if after == nil {
+		after = attrIDPrefix
+	}
+
+	k, _ = c.Seek(after)
+	if bytes.Equal(k, after) {
+		k, _ = c.Next()
+	}
+
+	for ; bytes.HasPrefix(k, attrIDPrefix); k, _ = c.Next() {
+		id, err := oid.DecodeBytes(k[len(attrIDPrefix):])
+		if err != nil {
+			return 0, nil, fmt.Errorf("wrong OID: %w", err)
+		}
+
+		var addr = oid.NewAddress(cnr, id)
+		hdr, err := get(tx, addr, oidKey, false, false, 0)
+		if err != nil {
+			return 0, nil, fmt.Errorf("header error for %s: %w", addr, err)
+		}
+
+		if hdr.HasParent() {
+			queue = append(queue, id)
+			if len(queue) == int(limit) {
+				break
+			}
+		}
+	}
+	for _, id := range queue {
+		_ = b.Delete(slices.Concat(attrIDPrefix, id[:]))
```

Comment:
why u decided to ignore error?

Reply:
it does not state this is the only possible error. Althought it is unexpected. Func returns an error, nothing breaks us to handle the case

Reply:
still do not log? even though added logger as an arg